### DISCUSSION TITLE
Airflow deployed on Kubernetes cluster NOT showing airflow app metrics in STATSD Exporter

### DISCUSSION THREAD

**Discussion by dibyo-gintaa** [apache/airflow#18556] [Q&A]
### Apache Airflow version

2.1.1

### Operating System

PRETTY_NAME="Debian GNU/Linux 10 (buster)" NAME="Debian GNU/Linux" VERSION_ID="10" VERSION="10 (buster)" VERSION_CODENAME=buster ID=debian

### Versions of Apache Airflow Providers

airflow:
  extraPipPackages: 
   - apache-airflow[statsd]

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

We have deployed the airflow as a helm release in a Kubernetes cluster.
So airflow web server is running as a pod in kubernetes cluster having statsd metrics in its values.yaml.

We have tried to follow this link: https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/metrics.html

We also deployed statsd exporter as a helm release.
Our flow is like this:    airflow app--> statsd config in values.yaml --> running statsd exporter where it will receive airflow metrics --> send the metrics to Prometheus server.
We have given the follow config in airflow values.yaml:
[metrics]
statsd_on = True
statsd_host = localhost
statsd_port = 8125
statsd_prefix = airflow

### What happened

We can not see any airflow metrics running in Host: 8125 or Host:125
When we do a local port forward of the statsd exporter pod we see only default statsd metric but not any airflow related metrics in statsd

### How to reproduce

1) Deploy airflow cluster as helm release using values.yaml
helm install poc-cluster-airflow  airflow-stable/airflow  --namespace airflow  --version 8.4.1  --values values-custom.yaml
values-custom.yaml having: 
 # Config settings to go into the mounted airflow.cfg
config:
  metrics:
    #statsd_on: '{{ ternary "True" "False" .Values.statsd.enabled }}'
    statsd_on: True
    statsd_port: 8125
    statsd_prefix: airflow
    statsd_host: '{{ printf "%s-statsd" .Release.Name }}'

2) helm --kubeconfig kube_config_cluster.yml install statsd-release prometheus-community/prometheus-statsd-exporter --namespace airflow

3) Try to port forward the statsd exporter and see metrics in localhost:9102/metrics. 
No airflow metrics is coming up

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by jedcunningham** [comment]
First off, it looks like you are using the user-community chart from https://github.com/airflow-helm/charts/tree/main/charts/airflow, not the official chart at https://airflow.apache.org/docs/helm-chart/stable/index.html.

The official chart already deploys a statsd-exporter by default. If you want to use the one that comes with the chart, it should work out of the box. Port forward to the `{.Release.Name}-statsd` service and you should see the metrics.

Alternatively, if you want to deploy your own exporter, you'd want this in your values:

```yaml
statsd:
  enabled: false
config:
  metrics:
    statsd_on: True
    statsd_host: statsd-release  # or whatever you name your prometheus-statsd-exporter service
```

**Reply by dibyo-gintaa** [reply]
Yes, I agree with you but we are actually getting the statsd metrics but we are looking for airflow dag metrics in it. So statsd metrics should show something like this as attached.  airflow-scheduler-tasks and all
The below statsd metrics I am getting but what is missing here is the airflow metrics:

airflow@poc-cluster-airflow-web-66877978db-2c6fq:/opt/airflow$ curl localhost:9102/metrics
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 2.7573e-05
go_gc_duration_seconds{quantile="0.25"} 4.3869e-05
go_gc_duration_seconds{quantile="0.5"} 5.2614e-05
go_gc_duration_seconds{quantile="0.75"} 7.3172e-05
go_gc_duration_seconds{quantile="1"} 0.017977237
go_gc_duration_seconds_sum 0.376161947
go_gc_duration_seconds_count 1700
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 14
# HELP go_info Information about the Go environment.
# TYPE go_info gauge
![af](https://user-images.githubusercontent.com/68272438/136325771-27e4be42-bd67-4bda-8422-442cf3b88911.png)

**Comment by dibyo-gintaa** [comment]
I have used official chart apache/airflow.     helm --kubeconfig kube_config_cluster.yml upgrade --install poc-cluster-airflow  apache-airflow/airflow  --namespace airflow  --values values.yaml       
statsd pod:        
 poc-cluster-airflow-statsd-87bf7f6b9-4q7vt               1/1     Running                 0          72m

On local port -forwarding the pod at 9102 port:  We are seeing same old metrics and no airflow dags metrics..

![image](https://user-images.githubusercontent.com/68272438/136511682-9cd8b855-ecf9-4d36-9b6c-988a37d64a74.png)

**Reply by rodrigo-l-santos-alb** [reply]
Hello, I am currently running into the same issue. I can only see statds metrics and do not see any of the airflow metrics. Were you able to find a solution, or was this fixed, and if so where can I find the fix?

**Reply by pcking999** [reply]
same issue just a basic default helm chart.
